inference/aot_serialization.md

```diff
@@ -0,0 +1,236 @@
+        "img_ids": torch.randn(num_img_sequences, 3, dtype=torch.bfloat16, device="cuda"),
+        "txt_ids": torch.randn(512, 3, dtype=torch.bfloat16, device="cuda"),
+        "guidance": torch.tensor([3.5],  device="cuda").expand(batch_size),
+        "return_dict": False,
+    }
+    return example_inputs
+
+def load_model():
+    model = FluxTransformer2DModel.from_pretrained(
+        "black-forest-labs/FLUX.1-dev", subfolder="transformer", torch_dtype=torch.bfloat16
+    ).to("cuda")
+    return model
+
+def benchmark_fn(f, *args, **kwargs):
+    t0 = benchmark.Timer(
+        stmt="f(*args, **kwargs)",
+        globals={"args": args, "kwargs": kwargs, "f": f},
+        num_threads=torch.get_num_threads(),
+    )
+    return f"{(t0.blocked_autorange().mean):.3f}"
+
+@torch.no_grad()
+def f(model, **kwargs):
+    return model(**kwargs)
+
+model = load_model()
+num_channels_latents = model.config.in_channels // 4 
+inputs = get_example_inputs(batch_size=1, height=1024, width=1024, num_channels_latents=num_channels_latents)
+
+for _ in range(5):
+    _ = f(model, **inputs)
+
+model = torch.compile(model, mode="max-autotune")
+
+time = benchmark_fn(f, model, **inputs)
+print(time) # 1.776 seconds on A100.
+```
+
+</details>    
+
+## AoT `torch.compile()`
+
+First, we obtain the AoT compiled `.so` file:
+
+<details>
+</summary>Code</summary>
+    
+```python
+import torch
+from diffusers import FluxPipeline, FluxTransformer2DModel
+import torch.utils.benchmark as benchmark
+from functools import partial
+
+def get_example_inputs():
+    example_inputs = {
+        "hidden_states": torch.load("latents.pt", map_location="cuda"),
+        "encoder_hidden_states": torch.load("prompt_embeds.pt", map_location="cuda"),
+        "pooled_projections": torch.load("pooled_prompt_embeds.pt", map_location="cuda"),
+        "timestep": torch.load("timestep.pt", map_location="cuda") / 1000,
+        "img_ids": torch.load("latent_image_ids.pt", map_location="cuda"),
+        "txt_ids": torch.load("text_ids.pt", map_location="cuda"),
+        "guidance": torch.load("guidance.pt", map_location="cuda"),
+        "joint_attention_kwargs": None, 
+        "return_dict": False
+    }
+    return example_inputs
+
+def benchmark_fn(f, *args, **kwargs):
+    t0 = benchmark.Timer(
+        stmt="f(*args, **kwargs)",
+        globals={"args": args, "kwargs": kwargs, "f": f},
+        num_threads=torch.get_num_threads(),
+    )
+    return f"{(t0.blocked_autorange().mean):.3f}"
+
+def load_model():
+    model = FluxTransformer2DModel.from_pretrained(
+        "black-forest-labs/FLUX.1-dev", subfolder="transformer", torch_dtype=torch.bfloat16
+    ).to("cuda")
+    return model
+
+def aot_compile(name, fn, **sample_kwargs):
+    path = f"{name}.so"
+    print(f"{path=}")
+    options = {
+        "aot_inductor.output_path": path,
+        "max_autotune": True,
+        "triton.cudagraphs": True,
+    }
+
+    torch._export.aot_compile(
```

Comment:
what's the plan for using the new api? `torch._inductor.aoti_compile_and_package`, do you want to update this after Bin fixed the model size issue?

Reply:
Yeah exactly. The gist that tackles the quantization + AoT part still uses `torch._inductor.aoti_compile_and_package`, though.